enterprise/backend/src/metabase_enterprise/sandbox/api/table.clj

```diff
@@ -28,14 +27,15 @@
                            [:= :pgm.user_id (u/the-id user-or-user-id)]]}))
 
 (mu/defn only-sandboxed-perms? :- :boolean
-  "Returns true if the user has only segemented and not full table permissions. If the user has full table permissions
-  we wouldn't want to apply this segment filtering."
+  "Returns true if the user has sandboxed permissions. If a sandbox policy exists, it overrides existing permission on
+  the table."
```

Comment:
There's a small distinction here between a sandboxing policy existing, and sandboxing actually being enforced for the user. Sandboxing is only enforced if the user isn't in any _other_ group which grants full access to the table.

It looks like the original function was checking whether sandboxing is enforced, while the new one is just checking whether any sandboxing policy exists.

I'm not sure if this distinction matters much for this namespace but it would definitely matter for the usages in `metabase-enterprise.sandbox.models.params.field-values` since it computes a different hash key for cached field values based on whether sandboxing is enforced or not.

To match the previous behavior you'd need to use `metabase-enterprise.sandbox.api.util/enforced-sandboxes`, or at least do something similar here.

Reply:
Thanks for noticing this and calling it out. I think I mentioned this in slack, but I had been thinking that our move from "sandbox is its own permission" to "sandbox is unrestricted perm + a GTAP" meant that we didn't need to check for unrestricted perms anymore, but as you pointed out, that's wrong - we need to check for unrestricted perms from _other_ groups.

I ended up modifying the `sandbox.api.util` namespace to add an `enforced-sandboxes-for` function. It takes a user ID and returns the sandboxes that will be enforced for that user. Then this function checks to see whether those enforced sandboxes include one for this table.

Reply:
Cool, makes sense!